src/Categories/Diagram/Pullback/Properties.agda

```diff
@@ -189,3 +189,130 @@ module IsoPb {X Y Z} {f : X ⇒ Z} {g : Y ⇒ Z} (pull₀ pull₁ : Pullback f g
 
   p₂-≈ : p₂ pull₁ ∘ P₀⇒P₁ ≈ p₂ pull₀
   p₂-≈ = p₂∘universal≈h₂ pull₁ {eq = commute pull₀}
+
+
+-- pasting law for pullbacks:
+-- in a commutative diagram of the form
+-- A -> B -> C
+-- |    |    |
+-- D -> E -> F
+-- if the right square (BCEF) is a pullback,
+-- then the left square (ABDE) is a pullback
+-- iff the big square (ACDF) is a pullback.
+module PullbackPastingLaw {A B C D E F : Obj} (f : A ⇒ B) (g : B ⇒ C) (h : A ⇒ D) (i : B ⇒ E) (j : C ⇒ F) (k : D ⇒ E) (l : E ⇒ F) (ABDE : i ∘ f ≈ k ∘ h) (BCEF : j ∘ g ≈ l ∘ i) (pbᵣ : IsPullback g i j l) where
+  open IsPullback using (p₁∘universal≈h₁; p₂∘universal≈h₂; universal; unique-diagram)
+  
+  ACDF : j ∘ (g ∘ f) ≈ (l ∘ k) ∘ h
+  ACDF = begin
+    j ∘ g ∘ f ≈⟨ sym-assoc ⟩
+    (j ∘ g) ∘ f ≈⟨ BCEF ⟩∘⟨refl ⟩
+    (l ∘ i) ∘ f ≈⟨ assoc ⟩
+    l ∘ (i ∘ f) ≈⟨ refl⟩∘⟨ ABDE ⟩
+    l ∘ (k ∘ h) ≈⟨ sym-assoc ⟩
+    (l ∘ k) ∘ h ∎
+
+  leftPullback⇒bigPullback : IsPullback f h i k → IsPullback (g ∘ f) h j (l ∘ k)
+  leftPullback⇒bigPullback pbₗ = record
+     { commute = ACDF
+     ; universal = universalb
+     ; p₁∘universal≈h₁ = [g∘f]∘universalb≈h₁
+     ; p₂∘universal≈h₂ = h∘universalb≈h₂
+     ; unique-diagram = unique-diagramb
+     } where
+        j∘h₁≈l∘k∘h₂ : {H : Obj} {h₁ : H ⇒ C} {h₂ : H ⇒ D} → j ∘ h₁ ≈ (l ∘ k) ∘ h₂ → j ∘ h₁ ≈ l ∘ (k ∘ h₂)
+        j∘h₁≈l∘k∘h₂ {H} {h₁} {h₂} eq = begin
+          j ∘ h₁ ≈⟨ eq ⟩
+          (l ∘ k) ∘ h₂ ≈⟨ assoc ⟩
+          l ∘ (k ∘ h₂) ∎
+                  
+        universalᵣ[h₁,k∘h₂] : {H : Obj} {h₁ : H ⇒ C} {h₂ : H ⇒ D} → j ∘ h₁ ≈ (l ∘ k) ∘ h₂ → H ⇒ B
+        universalᵣ[h₁,k∘h₂] {H} {h₁} {h₂} eq = universal pbᵣ {H} {h₁} {k ∘ h₂} (j∘h₁≈l∘k∘h₂ eq)
```

Comment:
in the definition (line 229, not 228) can you figure out which of the implicits can be omitted, if any? You can use named bindings to only give the ones necessary.  Here and through the rest of the file please.